RELEASE-PROCEDURE: update to new schedule (#10827)
curl/curl: docs/RELEASE-PROCEDURE.md

Release Cycle
-------------

> We normally do releases every 8 weeks on Wednesdays. If critical problems

Reviewer: "critical" is now too strong.

> arise, we can insert releases outside of the schedule or we can move the

Reviewer: Drop "of"

release date.

> Each 8 week (56 days) release cycle is split in three distinct periods.

Reviewer 2: ```suggestion
Each 8 week (56 days) release cycle is split into three distinct periods.
```

or "divided by" ?

> - During the first calendar 10 days after a release, we are in "cool down". We

Reviewer 3: Should this be `10 calendar days` instead of `calendar 10 days`?

> - During the first calendar 10 days after a release, we are in "cool down". We

Reviewer 2: ```suggestion
- During the first 10 calendar days after a release, we are in "cool down". We
```

>   do not merge features but only bug-fixes. If any important enough regression

Reviewer 2: ```suggestion
  do not merge features but only bug-fixes. If a critical regression
```

  reported, we might do a follow-up patch release.

- During the following 3 weeks (21 days) there is a feature window: we allow
  new features and changes to curl and libcurl. If we accept any such changes,
  we bump the minor number used for the next release.

- During the next 25 days we are in feature freeze. We do not merge any

>   features or changes, we then only focus on fixing bugs and polishing things

Reviewer 2: ```suggestion
  features or changes, and we only focus on fixing bugs and polishing things
```

  to make a solid coming release.

If a future release date happens to end up on a "bad date", like in the middle

> of common public holidays or when the lead release manager is away traveling,

Reviewer 2: ```suggestion
of common public holidays or when the lead release manager is unavailable,
```
